### PR TITLE
[RSDK-4423] - Obstacle avoidance UX improvements and bugfixes

### PR DESCRIPTION
**Pull request by micheal-parks**
In this PR:
- Filter NaN values from `navigationService.getLocation` - This is an API issue and needs to be fixed upstream eventually
- Show waypoint tab by default
- Introduce persisted store to save tab selection in localStorage
- Show obstacle rotations in obstacle list for debugging
- Correctly adapt AxesHelper sizes to individual obstacle size based on bounding spheres
- Zoom to obstacles based on bounding sphere size when selected

### CODE REVIEW

File: web/frontend/src/api/navigation.ts [hunk 1]
```diff
@@ -198,13 +198,14 @@ export const getLocation = async (robotClient: Client, name: string) => {
...
   const lng = location?.getLongitude();
 
-  rcLogConditionally(response?.getLocation()?.toObject());
-  rcLogConditionally(location?.getLatitude());
-  rcLogConditionally(location?.getLongitude());
-
-  if (typeof lat !== 'number' || typeof lng !== 'number') {
-    // eslint-disable-next-line unicorn/prefer-type-error
-    throw new Error('Unable to locate robot');
+  // todo(micheal parks) - This should be abstracted into the TS SDK, returning NaN here is non-typical
```
Comment (micheal-parks) [inline web/frontend/src/api/navigation.ts:201]: @maxhorowitz @purplenicole730 FYI

File: web/frontend/src/components/navigation/components/nav/obstacles.svelte [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 import OrientationInput from '../input/orientation.svelte';
-import { obstacles, flyToMap, write, hovered, mapCenter } from '../../stores';
+import { obstacles, write, hovered, mapCenter, zooms, flyToMap } from '../../stores';
 import { createObstacle } from '../../lib/obstacle';
 import type { Geometry, LngLat } from '@/api/navigation';
 
-const dispatch = createEventDispatcher<{ select: LngLat }>();
+const handleSelect = (selection: { name: string; location: LngLat }) => {
+  const zoom = zooms[selection.name];
+  console.log(zoom);
```
Comment (ethanlook) [inline web/frontend/src/components/navigation/components/nav/obstacles.svelte:12]: Errant console log

File: web/frontend/src/stores/persisted.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * @param key The storage key
+ * @param initialValue The initial value to put in storage if no value exists.
+ * @param storageType 'local' or 'session'
+ */
+export const persisted = <T>(
+  key: string,
+  initialValue: T,
+  storageType: 'local' | 'session' = 'local'
+): Writable<T> => {
+  const browser = typeof window !== 'undefined' && typeof document !== 'undefined';
```
Comment (ethanlook) [inline web/frontend/src/stores/persisted.ts:25]: Should we leave a TODO to use the SvelteKit browser flag?

**Review by ethanlook — APPROVED** [review]

File: web/frontend/src/stores/persisted.ts (hunk 3 above)
Comment (micheal-parks) [reply web/frontend/src/stores/persisted.ts:25]: Good idea, will add

**Pull request merged**